### PR TITLE
Remove mention of tier one repos in documentation

### PR DESCRIPTION
**Pull request by riarenas**
We no longer have a list of tier 1 repos.
fixes https://github.com/dotnet/dnceng/issues/1023

### To double check:

* [ ] The right tests are in and and the right validation has happened.  Guidance:  https://github.com/dotnet/arcade/tree/main/Documentation/Validation

### CODE REVIEW

File: Documentation/ArcadeSdk.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 Arcade SDK is a set of msbuild props and targets files and packages that provide common build features used across multiple repos, such as CI integration, packaging, VSIX and VS setup authoring, testing, and signing via Microbuild.
 
-The infrastructure of each [repository that contributes to .NET Core 3.0 stack](TierOneRepos.md) is built on top of Arcade SDK. This allows us to orchestrate the build of the entire stack as well as build the stack from source. These repositories are expected to be on the latest version of the Arcade SDK.
+The infrastructure of each repository that contributes to .NET Core 3.0 stack is built on top of Arcade SDK. This allows us to orchestrate the build of the entire stack as well as build the stack from source. These repositories are expected to be on the latest version of the Arcade SDK.
```
Comment (mmitche) [inline Documentation/ArcadeSdk.md:5]:
```suggestion
The infrastructure of most repositories that contribute to the .NET stack is built on top of Arcade SDK. This allows us to orchestrate the build of the entire stack as well as build the stack from source. These repositories are expected to be on the latest version of the Arcade SDK.
```

**Review by mmitche — DISMISSED** [review]

File: Documentation/ArcadeSdk.md [hunk 2]
```diff
@@ -2,7 +2,7 @@
...
 
 Repositories that do not participate in .NET Core 3.0 build may also use Arcade SDK in order to take advantage of the common infrastructure.
```
Comment (dougbu) [inline Documentation/ArcadeSdk.md:7]: similarly
```suggestion
Repositories that do not contribute to the .NET stack may also use Arcade SDK in order to take advantage of the common infrastructure.
```

**Review by dougbu — DISMISSED** [review]

**Comment by dougbu** [comment]
Should we take the TierOneRepos.md removal and changes here to the servicing branches❔ Alternatively, are the servicing branches candidates for `git rm -r ./Documentation`❔

**Comment by riarenas** [comment]
> Alternatively, are the servicing branches candidates for git rm -r ./Documentation❔

I think I like this idea. We don't really maintain the docs in the other branches afaik.

**Review by missymessa — APPROVED** [review]
LGTM shipit

**Comment by dougbu** [comment]
> > Alternatively, are the servicing branches candidates for git rm -r ./Documentation❔
> 
> I think I like this idea. We don't really maintain the docs in the other branches afaik.

Don't think we have this issue. I'll file something and we can let anyone who objects have their say.

**Comment by dougbu** [comment]
^^ now immortalized as #14152

**Pull request merged**